### PR TITLE
MEN-1188 Force terminating hanging scripts.

### PR DESCRIPTION
**Pull request by pasinskim**
Force termination of state scripts that are hanging.

Changelog: None

Signed-off-by: Marcin Pasinski <marcin.pasinski@mender.io>

### CODE REVIEW

File: statescript/executor.go [hunk 1]
```diff
@@ -102,6 +101,41 @@ func execute(name string) int {
...
+	cmd := exec.Command(name)
+	if err := cmd.Start(); err != nil {
+		return retCode(err)
+	}
+	// As chaild process gets the same PGID as the parent by default, in order
+	// to avoid killing Mender when killing process group we are setting
+	// new PGID for the executed script and its children.
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
+	timer := time.AfterFunc(timeout, func() {
```
Comment (kacf) [inline statescript/executor.go]: That's really handy!

File: statescript/executor.go [hunk 2]
```diff
@@ -102,6 +101,41 @@ func execute(name string) int {
...
+	return t
+}
+
+func execute(name string, timeout time.Duration) int {
+
+	cmd := exec.Command(name)
+	if err := cmd.Start(); err != nil {
+		return retCode(err)
+	}
+	// As chaild process gets the same PGID as the parent by default, in order
```
Comment (bboozzoo) [inline statescript/executor.go]: child

File: statescript/executor.go [hunk 3]
```diff
@@ -102,6 +101,41 @@ func execute(name string) int {
...
+	// to avoid killing Mender when killing process group we are setting
+	// new PGID for the executed script and its children.
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
```
Comment (bboozzoo) [inline statescript/executor.go]: Don't you need to set these before calling cmd.Start() ?

File: statescript/executor.go (hunk 2 above)
Comment (pasinskim) [inline statescript/executor.go]: Fixed!

File: statescript/executor.go (hunk 3 above)
Comment (pasinskim) [inline statescript/executor.go]: Right. Thanks for spotting!

**Comment by pasinskim** [comment]
After rebasing both this one and integration https://github.com/mendersoftware/integration/pull/262 build was green (or actually blue): https://mender-jenkins.mender.io/job/mender-builder/931

**Pull request merged**